cli.go

```diff
@@ -58,96 +58,91 @@ func (h *CLIHandler) Auth(c *api.Client, m map[string]string) (*api.Secret, erro
+	authType, ok := m["auth_type"]
 	if !ok {
-		return nil, fmt.Errorf("Enter the auth_type")
+		return nil, fmt.Errorf("'auth_type' is required")
 	}
 
-	switch strings.ToLower(authtype) {
+	var headerFunc func(string, string) (http.Header, error)
+	switch strings.ToLower(authType) {
 	case "ip", "instance":
-		return createLoginDataForInstancePrincipal(clientAddress, path)
+		headerFunc = GetSignedInstanceRequestHeaders
 	case "ak", "apikey":
-		return createLoginDataForApiKeys(clientAddress, path)
+		headerFunc = GetSignedAPIRequestHeaders
+	default:
+		return nil, fmt.Errorf("unsupported auth_type %q", authType)
 	}
 
-	return nil, fmt.Errorf("Unknown auth_type")
-}
-
-func createLoginDataForApiKeys(clientAddress string, path string) (map[string]interface{}, error) {
-
-	provider := common.DefaultConfigProvider()
-
-	ociClient, err := NewOciClientWithConfigurationProvider(provider)
+	headers, err := headerFunc(addr, path)
 	if err != nil {
 		return nil, err
 	}
 
-	return createFinalLoginData(clientAddress, &ociClient, path)
+	return map[string]interface{}{
+		"request_headers": headers,
+	}, nil
 }
 
-func createLoginDataForInstancePrincipal(clientAddress string, path string) (map[string]interface{}, error) {
-
+func GetSignedInstanceRequestHeaders(addr, path string) (http.Header, error) {
 	ip, err := auth.InstancePrincipalConfigurationProvider()
 	if err != nil {
 		return nil, err
 	}
-	ociClient, err := NewOciClientWithConfigurationProvider(ip)
+
+	c, err := NewOciClientWithConfigurationProvider(ip)
 	if err != nil {
 		return nil, err
 	}
-	return createFinalLoginData(clientAddress, &ociClient, path)
+	return getSignedRequestHeaders(addr, &c, path)
 }
 
-func createFinalLoginData(clientAddress string, ociClient *OciClient, path string) (map[string]interface{}, error) {
-
-	ociClient.Host = clientAddress
-	request, err := ociClient.ConstructLoginRequest(path)
+func GetSignedAPIRequestHeaders(addr, path string) (http.Header, error) {
+	c, err := NewOciClientWithConfigurationProvider(common.DefaultConfigProvider())
 	if err != nil {
 		return nil, err
 	}
 
-	clientURL, err := url.Parse(clientAddress)
+	return getSignedRequestHeaders(addr, &c, path)
+}
+
+func getSignedRequestHeaders(addr string, client *OciClient, path string) (http.Header, error) {
+	clientURL, err := url.Parse(addr)
 	if err != nil {
 		return nil, err
 	}
-	request.Host = clientURL.Host
-
-	// serialize the request
-	serializedRequest := serializeRequest(request)
 
-	// pack it into loginData
-	loginData := make(map[string]interface{})
-	loginData["request_headers"] = serializedRequest
+	client.Host = addr
+	request, err := client.ConstructLoginRequest(path)
+	if err != nil {
+		return nil, err
+	}
 
-	return loginData, nil
-}
+	request.Host = clientURL.Host
+	request.Header.Set("host", request.Host)
+	request.Header.Set("(request-target)",
+		fmt.Sprintf("%s %s", strings.ToLower(request.Method), request.URL.RequestURI()))
```

Comment:
Why don't we use the `getRequestTarget` util here?

Reply:
Good call, I had forgotten to drop that function. It was only ever used here.

Fixed with: e04f9962bf19d79b8da47ae3f1a1ef4c2b6f5e78